### ISSUE TITLE
[Faceted Filters] Mobile/Drawer facet launcher and container missing attributes for context

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#119]
## Issue summary

The mobile/Drawer facet launcher and container missing attributes to convey context. The current aural user experience offering produces semantics for a [disclosure pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure) with a mix of keyboard support for a modal context. The visually user experience portrays a modal window pattern. This UI mismatch could cause issues for sighted, low-vision, and blind screen reader users.

<details>
  <summary>Screenshots</summary>
<img width="416" alt="Screenshot of Dawn mobile facet content drawer. Drawer container is highlighted." src="https://user-images.githubusercontent.com/1392632/124609384-9212a980-de3d-11eb-9765-b3a19b1523e0.png">

</details>

#### HTML

```html
<menu-drawer class="mobile-facets__wrapper">
  <details
    class="disclosure-has-popup medium-hide large-up-hide menu-opening"
    open
  >
    <summary role="button" aria-expanded="true" aria-controls="">
      <!-- … -->
    </summary>
    <collection-filters-form>
      <form id="CollectionFiltersFormMobile" class="mobile-facets">
        <div class="mobile-facets__inner">
          <div class="mobile-facets__header">
            <div class="mobile-facets__header-inner">
              <h2 class="mobile-facets__heading">Filter and sort</h2>
              <!-- … -->
            </div>
          </div>
        </div>
      </form>
    </collection-filters-form>
  </details>
</menu-drawer>
```

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported mobile browser.
2. With an active screen reader, navigate to and activate the Filter and Sort control.

### Expected

- Activator and container window to be conveyed as anf behave like a modal window.

### Actual

- Activator and container window are conveyed as disclosure pattern.

## Recommendation

**In the JS env only:**

**Launcher:**

1. Remove the `aria-expanded` and `aria-controls` from the `summary` activator as these are not required for a modal pattern.
2. Add the `aria-describedby` attribute to the `summary` activator, setting its value to `a11y-modal-window-message` (via https://github.com/Shopify/dawn/issues/1446.)

**Close control:**

1. Set the current `.mobile-facets__close` as `display:none` – the close control must be **within** the modal context for it to be reachable (described below.)
2. Include a new close control `button` in the DOM within the `.mobile-facets__header` container, providing an appropriate accessible name.

**Content container:**

1. Wrap the `#CollectionFiltersFormMobile` element with a new `div` container.
2. Add the `role` attribute to the `div`, setting its value to `dialog`. This conveys the presence of the modal window to screen reader users.
3. Add the `aria-labelledby` attribute to the `div`, setting its value to match the `id` of the container `h2` heading. This give the modal window an accessible name.
4. Add the `aria-modal` attribute to the `div`, setting its value to `true`. This conveys the presence of the modal window to assistive technology.
5. Add the `tabindex` attribute to the `div`, setting its value to `-1`. This allows the modal window to receive keyboard focus upon activation.

**Focus management:**

1. When the modal window launcher is activated, shift keyboard focus to the `div` container. This will alert the user to the new modal context.
2. When the modal window is dismissed via Close `button` control, "Clear" control, "Apply" control, or <kbd>Esc</kbd> key press, shift focus back to the launcher control. This allows the user to navigate forward from where they left off.

**This also applies to the Drawer filter layout.**

#### HTML

```html
<menu-drawer class="mobile-facets__wrapper">
  <details
    class="disclosure-has-popup medium-hide large-up-hide menu-opening"
    open
  >
    <summary role="button" aria-describedby="a11y-modal-window-message">
      <!-- … -->
    </summary>
    <collection-filters-form>
      <div
        role="dialog"
        aria-labelledby="MobileFacetsHeading"
        aria-modal="true"
        tabindex="-1"
      >
        <form id="CollectionFiltersFormMobile" class="mobile-facets">
          <div class="mobile-facets__inner">
            <div class="mobile-facets__header">
              <div class="mobile-facets__header-inner">
                <h2 class="mobile-facets__heading" id="MobileFacetsHeading">
                  Filter and sort
                </h2>
                <!-- … -->
                <button
                  type="button"
                  class="mobile-facets__close mobile-facets__close--no-js"
                  aria-label="Close"
                >
                  <!-- SVG… -->
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </collection-filters-form>
  </details>
</menu-drawer>
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Operable, Robust
- **WCAG SC:** [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html), [4.1.2  Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- **Severity:** High
- **Effort:** High
<!-- - **Related:** -->
